### PR TITLE
Clear categorical and numerical bins when clearing high dimensional nodes

### PR DESCRIPTION
**Pull request by kretep**

### CODE REVIEW

File: web-app/js/plugin/BoxPlot.js [hunk 1]
```diff
@@ -370,5 +370,43 @@ BoxPlotView.prototype.submit_job = function (form) {
 
 }
 
+BoxPlotView.prototype.clear_high_dimensional_input = function (div) {
+    RmodulesView.prototype.clear_high_dimensional_input.call(this, div);
```
Comment (forus) [inline web-app/js/plugin/BoxPlot.js]: Wouldn't `RmodulesView.prototype.clear_high_dimensional_input.call(this, div);` work?

File: web-app/js/plugin/BoxPlot.js [hunk 2]
```diff
@@ -370,5 +370,43 @@ BoxPlotView.prototype.submit_job = function (form) {
...
+
+    // Only clear bins for corresponding variable (dependent/independent)
+    var variableToBeBinned = document.getElementById("selBinVariableSelection").value
+    if (div == "divDependentVariable" && variableToBeBinned == "DEP" ||
```
Comment (forus) [inline web-app/js/plugin/BoxPlot.js]: Shouldn't you check wether binning is enabled(is checkbox selected)?

File: web-app/js/plugin/BoxPlot.js (hunk 1 above)
Comment (kretep) [inline web-app/js/plugin/BoxPlot.js]: I don't think so. The function is defined in the prototype. RmodulesView is not an object, but the "super class" of BoxPlotView.

File: web-app/js/plugin/BoxPlot.js (hunk 2 above)
Comment (kretep) [inline web-app/js/plugin/BoxPlot.js]: I think the boxes should be cleared either way, even if binning is disabled. If the user enables binning after clearing, you don't want to see the previously cleared categories, right?

**Pull request merged**